docs/configuration_files.rst

```diff
@@ -122,3 +122,264 @@ An example of the main configuraiton file:
+
+.. _flat_pattern:
+
+Flat Pattern
+------------
+
+List of strings that will be used to identify  flat field files within the list of input files. mkrefs will look for these strings in the filenames.
+
+.. _basenamepattern:
+
+Basename Pattern
+----------------
+
+String to use when looking for input files.
+
+.. _dateobs_fitskey:
+
+Date-Obs Fits Header Keyword
+----------------------------
+
+The header keyword in the input files that contains the date of the observation. Dates are needed to enforce rules when pairing darks and flats. For JWST, set this to **DATE-END**.
+
+.. _timeobs_fitskey:
+
+Time-Obs Fits Header Keyword
+----------------------------
+
+The header keyword in the input files that contains the time of the observation. Time are needed to enforce rules when pairing darks and flats. For JWST, set this to **TIME-END**.
+
+.. _mjdobs_fitskey:
+
+Obervation MJD Fits Header Keyword
+----------------------------------
+
+The header keyword in the input files that contains the time of the observation in MJD.
+
+.. _requiredfitskeys:
+
+Required Fits File Header Keywords
+----------------------------------
+
+List of header keywords which must be present in the input files.
+
+.. _optionalfitskeys:
+
+Optional FitsFile Header Keywords
+---------------------------------
+
+List of optional header keywords in the input files. Used for
+
+.. _DD:
+
+DD
+---
+
+Stands for dark dark, and indicates a situation where a pair of dark current ramps are needed to produce an individual reference file.
+
+.. _FF:
+
+FF
+---
+
+Stands for flat flat, and indicates a situation where a pair of flat field ramps are needed to produce an individual reference file.
+
+.. _DDFF:
+
+DDFF
+----
+
+Stands for dark dark flat flat, and indicates a situation where a pair of dark current ramps and a pair of flat field ramps are needed to produce an individual reference file.
+
+.. _max_Delta_MJD:
+
+Max_Delta_MJD
+-------------
+The maximum time allowed, in days, between input observations when creating pairs/groups of files. For example, to minimize the chances of dark current varying enough to impact reference file creation, you can set `max_Delta_MJD` to (e.g.) 2. In this case, when pairing dark current files (i.e. DD_) *jwst_reffiles* will not pair observations taken more than 2 days apart.
+
+.. _basename:
+
+Reference File Output Basename
+------------------------------
+
+Format of the output names for individual reference files. Output names will be automatically generated by *jwst_reffiles* to ensure accurate bookkeeping. The overall format of the reference file output names follows the convention:
+
+outrootdir_[/outsubdir_][/runID_]/reflabel/reflabel[_outsubdir_][_runID_][.addsuffix_].cmdID.reftype_.fits
+
+.. _outrootdir:
+
+Output Root Directory
+---------------------
+
+Path to the top level output directory for *jwst_reffiles*. Default is to define this within the **JWST_MKREFS_OUTDIR** environment variable, but any valid path is acceptable.
```

Comment:
if it's not defined, does the code fail or is it just written to e.g. cwd?

Reply:
#46 is tracking this. A check needs to be added to mkrefs.py that the output dir is a valid path. There is error checking for the case where it is left undefined.